src/main/scala-2.12/org/apache/james/gatling/smtp/SmtpProtocolBuilder.scala

```diff
@@ -1,25 +1,19 @@
 package org.apache.james.gatling.smtp
 
 object SmtpProtocolBuilder {
-
   val default = new SmtpProtocolBuilder(SmtpProtocol.default)
-
 }
 
 case class SmtpProtocolBuilder(protocol: SmtpProtocol) {
 
   def host(host: String) = copy(protocol.copy(host = host))
   def ssl(ssl: Boolean) = copy(protocol.copy(ssl = ssl))
   def port(port: Int) = copy(protocol.copy(port = port))
-  def port(port: String) = if (port != null) copy(protocol.copy(port = Integer.valueOf(port))) else this
+  def port(port: String) = Option(port).map(_ -> copy(protocol.copy(port = Integer.valueOf(port)))).getOrElse(this)
```

Comment:
does it even make sense to support null values? I mean, Scala devs don't use null ...

Reply:
The protocol use a java Integer, not a scala Int. I think we should refactor it. And then we could drop the null handling.

Reply:
Int can be null, like Integer. And we don't use null in Java neither so the refactoring, while still fine, is not a prerequisite, don't you think?

Reply:
Yes i agree with you.

Reply:
I applied Remi's fixup about this, removing entirely the useless SmtpProtocolBuilder class